sdks/python/src/opik/evaluation/utils.py

```diff
@@ -0,0 +1,45 @@
+from typing import List
+
+from opik.api_objects import opik_client
+from opik.evaluation import test_case
+
+
+def get_trace_project_name(trace_id: str) -> str:
+    # We first need to get the project_id for the trace
+    client = opik_client.get_client_cached()
+    traces = client._rest_client.traces.get_trace_by_id(id=trace_id)
+    project_id = traces.project_id
+
+    # Then we can get the project name
+    project_metadata = client._rest_client.projects.get_project_by_id(id=project_id)
+    return project_metadata.name
+
+
+def get_experiment_test_cases(
+    client: opik_client.Opik, experiment_id: str, dataset_id: str
+) -> List[test_case.TestCase]:
+    test_cases = []
+    page = 1
+
+    while True:
```

Comment:
@alexkuzmik I tried to do this but I couldn't get it to work, the stream endpoint doesn't return the input or output just the trace_id. I think tried to move the logic I have using the FE endpoint but that requires a dataset_id rather than dataset_name which made the whole logic a lot more complicated

So for now I recommend we keep it as is and come back later to clean up the code